test/Microsoft.Extensions.Options.Test/OptionsBuilderTest.cs

```diff
@@ -387,15 +387,29 @@ public ValidateOptionsResult Validate(string name, ComplexOptions options)
             }
         }
 
-        private void ValidateFailure<TOptions>(OptionsValidationException e, string name = "", params string[] errors)
+        private void ValidateFailure<TOptions>(OptionsValidationException e, string name = "", int count = 1, params string[] errorsToMatch)
         {
             Assert.Equal(typeof(TOptions), e.OptionsType);
             Assert.Equal(name, e.OptionsName);
-            if (errors.Length == 0)
+            if (errorsToMatch.Length == 0)
             {
-                errors = new string[] { "A validation error has occured." };
+                errorsToMatch = new string[] { "A validation error has occured." };
+            }
+            Assert.Equal(count, e.Failures.Count());
+            // Check for the error in any of the failures
+            foreach (var error in errorsToMatch)
+            {
+                var found = false;
```

Comment:
I bet this whole inner loop can be one LINQ statement. Might be easier to read. Up to you.

Reply:
Yeah good call, much more compact